torch/csrc/jit/mobile/model_tracer/OperatorCallTracer.h

```diff
@@ -0,0 +1,48 @@
+#pragma once
+
+#include <ATen/record_function.h>
+
+namespace torch {
+namespace jit {
+namespace mobile {
+/* The OperatorCallTracer class handles the attachment and removal of a
+ * recording callback that traces invocation of ATen (and other) PyTorch
+ * operators that get called via the Dispatcher.
+ *
+ * You can get the set of operators that were called (op_name.overload_name)
+ * using getCalledOperators().
+ *
+ * Note: This class is not thread safe or re-entrant, and should not be used
+ * across multiple threads of execution.
+ *
+ */
+struct OperatorCallTracer final {
+  static std::set<std::string> called_operators_;
+  at::CallbackHandle handle_;
+
+  OperatorCallTracer() {
```

Comment:
Is `OperatorCallTracer` supposed to be a singleton? If so, wouldn't it be better to make its constructor private?